### PR TITLE
LPS-141862 Trim left zero

### PR DESCRIPTION
**Pull request by alinedoleron**
Related Issue: https://issues.liferay.com/browse/LPS-141862

### CODE REVIEW

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @alinedoleron, can you please check if these failures are related to your changes? Thanks

**Review by miltonmc — CHANGES_REQUESTED** [review]
You did great!

Maybe be a good idea use recursion to remove more than one left zeros  on `trimLeftZero`, so that the function could be used to properly format pasted values like `000,1'

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/util/numericalOperations.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	return value;
+};
+
+export const trimLeftZero = ({
+	decimalSymbol,
+	thousandsSeparator,
+	value,
+}: {
+	decimalSymbol: string;
+	thousandsSeparator: any;
```
[the remark this answers is not in the document]
Comment (alinedoleron) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/util/numericalOperations.tsx]: @miltonmc, thank you for the suggestion of doing the recursion. So much simpler! You're a genius :sunglasses::clap:!!!

**Comment by alinedoleron** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/util/numericalOperations.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+		return defaultValue;
+	}
+	else if (value > max) {
+		return max;
+	}
+
+	return value;
+};
+
+export const trimLeftZero: any = ({
```
Comment (alinedoleron) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/util/numericalOperations.tsx:38]: @miltonmc, I put `any` here but I don't know if it's the best for this case and I don't know how to do :worried:

Comment (miltonmc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/util/numericalOperations.tsx:38]: Due to the recursion the TypeScript could not infer the return type properly, so we have to enforce the it:
```typescript
export const trimLeftZero: (options: {
	decimalSymbol: string;
	thousandsSeparator: string | void;
	value: string;
}) => string = ({
	decimalSymbol,
	thousandsSeparator,
	value,
}) => {
	...
};
```

**Review by miltonmc — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/util/numericalOperations.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (miltonmc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/util/numericalOperations.tsx]: > @miltonmc, thank you for the suggestion of doing the recursion. So much simpler! You're a genius sunglassesclap!!!

I'm happy you liked it, but  I'm not the genius you think I'm!

I got insights about recursion from those [classes](https://www.edx.org/course/introduction-to-computer-science-and-programming-7?index=product&queryID=d15b13a6cb0202b40baf8401bf2e68ea&position=1)... I'm strongly recommend it for everyone!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/util/numericalOperations.tsx (hunk 2 above)
Comment (alinedoleron) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/util/numericalOperations.tsx]: Thank you!!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/util/numericalOperations.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (alinedoleron) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/util/numericalOperations.tsx]: hahaha Yes, you are!! 
Awesome, I've just subscribed!! Thankss!!

**Comment by alinedoleron** [comment]
ci:stop

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @alinedoleron, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @alinedoleron, can you please check if these failures are related to your changes? Thanks

**Review by miltonmc — APPROVED** [review]
LGTM!

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @alinedoleron, can you please check if these failures are related to your changes? Thanks

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**